rcl/src/rcl/arguments.c

```diff
@@ -1074,6 +1162,77 @@ _rcl_parse_param_file_rule(
   return RCL_RET_INVALID_PARAM_RULE;
 }
 
+rcl_ret_t
+_rcl_parse_external_log_config_file(
+  const char * arg,
+  rcl_allocator_t allocator,
+  char ** log_config_file)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(arg, RCL_RET_INVALID_ARGUMENT);
+
+  const size_t param_prefix_len = sizeof(RCL_EXTERNAL_LOG_CONFIG_ARG_RULE);
+  if (strncmp(RCL_EXTERNAL_LOG_CONFIG_ARG_RULE, arg, param_prefix_len) == 0) {
+    size_t outlen = strlen(arg) - param_prefix_len;
+    log_config_file = allocator.allocate(sizeof(char) * (outlen + 1), allocator.state);
```

Comment:
Is `log_config_file` allowed to be `NULL`?

Also, shouldn't it be `*log_config_file = ...`? Unless I miss understand, isn't this supposed to be an output variable? Changing the value of `log_config_file` will not affect the the value passed into the function, which is an address in a struct anyways. Sorry if I'm "off-by-one" dereferencing there, this code is always tricky.

Reply:
The `log_config_file` input should be allowed to be NULL or an empty string if no config file is provided. 

You are correct about the pointer. A last minute change didn't get fully tested and this slipped by. I'll correct it.

Reply:
Fixed the pointer dereference.